File: app/src/main/java/com/example/rickmortymvvm/intrefaces/Observer.java
package com.example.rickmortymvvm.intrefaces;

import com.example.rickmortymvvm.list.viewmodel.PresentationCharacterListState;
import com.example.rickmortymvvm.list.viewmodel.PresentationCharacterListAction;

public interface Observer {
    void notify(PresentationCharacterListState state);

    void notify(PresentationCharacterListAction action);
}

File: app/src/main/java/com/example/rickmortymvvm/intrefaces/Observer.kt
package com.example.rickmortymvvm.intrefaces

import com.example.rickmortymvvm.list.viewmodel.PresentationCharacterListAction
import com.example.rickmortymvvm.list.viewmodel.PresentationCharacterListState

interface Observer {
    fun notify(state: PresentationCharacterListState)
    fun notify(action: PresentationCharacterListAction?)
}